toolchain/check/handle_class.cpp

```diff
@@ -174,6 +174,77 @@ auto HandleBaseColon(Context& /*context*/, Parse::NodeId /*parse_node*/)
   return true;
 }
 
+namespace {
+// Information gathered about a base type specified in a `base` declaration.
+struct BaseInfo {
+  SemIR::TypeId type_id;
+  SemIR::NameScopeId scope_id;
+
+  // A `BaseInfo` representing an erroneous base.
+  static const BaseInfo Error;
+};
+constexpr BaseInfo BaseInfo::Error = {.type_id = SemIR::TypeId::Error,
+                                      .scope_id = SemIR::NameScopeId::Invalid};
```

Comment:
Declaration order: static constants come early. FWIW you could also just make this a non-member.
https://google.github.io/styleguide/cppguide.html#Declaration_Order

```suggestion
// Information gathered about a base type specified in a `base` declaration.
struct BaseInfo {
  // A `BaseInfo` representing an erroneous base.
  static const BaseInfo Error;

  SemIR::TypeId type_id;
  SemIR::NameScopeId scope_id;
};
constexpr BaseInfo BaseInfo::Error = {.type_id = SemIR::TypeId::Error,
                                      .scope_id = SemIR::NameScopeId::Invalid};
```

```suggestion
// Information gathered about a base type specified in a `base` declaration.
struct BaseInfo {
  SemIR::TypeId type_id;
  SemIR::NameScopeId scope_id;
};
// A `BaseInfo` representing an erroneous base.
static constexpr BaseInfo BaseInfoError = {
    .type_id = SemIR::TypeId::Error, .scope_id = SemIR::NameScopeId::Invalid};
```